pwa/src/components/TestResultsForm.tsx

```diff
@@ -0,0 +1,134 @@
+                <CheckboxItem
+                    id="blood-sugar-test"
+                    data-testid="blood-sugar-test"
+                    onChange={() => setBloodSugarTest(bloodSugarTest ? 0 : 1)}
+                >
+                    Blood Sugar Test Performed
+                </CheckboxItem>
+                {bloodSugarTest ? (
+                    <div
+                        id="blood-sugar-results"
+                        data-testid="blood-sugar-results"
+                    >
+                        <List renderHeader={() => 'Blood Sugar Results'}>
+                            <InputItem
+                                type="number"
+                                id="blood-sugar-results-val"
+                                data-testid="blood-sugar-results-val"
+                                placeholder="mg/dL" // TOOD: unsure if the unit is mg/dL or mmol/L, could leave this out
+                                maxLength={3}
+                            ></InputItem>
+                            <WhiteSpace className="list-whitespace" size="lg" />
+                        </List>
+                    </div>
+                ) : null}
+                <CheckboxItem
+                    id="ua-test"
+                    data-testid="ua-test"
+                    onChange={() => setUA(ua ? 0 : 1)}
+                >
+                    UA (Urine Analysis) Test Performed
+                </CheckboxItem>
+                {ua ? (
+                    <div id="ua-results" data-testid="ua-results">
+                        <List renderHeader={() => 'UA Results'}>
+                            {/* TODO: I am unsure about the range/type of these fields - check DB */}
+                            <InputItem
+                                id="glucose-results-val"
+                                data-testid="glucose-results-val"
+                                placeholder="0"
+                            >
+                                Glucose
+                            </InputItem>
+                            <InputItem
+                                id="nitrites-results-val"
+                                data-testid="nitrites-results-val"
+                                placeholder="0"
+                            >
+                                Nitrites
+                            </InputItem>
+                            <InputItem
+                                id="protein-results-val"
+                                data-testid="protein-results-val"
+                                placeholder="0"
+                            >
+                                Protein
+                            </InputItem>
+                            <InputItem
+                                id="leukocytes-results-val"
+                                data-testid="leukocytes-results-val"
+                                placeholder="0"
+                            >
+                                Leukocytes
```

Comment:
duplicates could be made into for loop

pseudocode of idea:
```
for r in result_list:
<InputItem
    id="{r}-results-val"
    data-testid="{r}-results-val"
    placeholder="0"
> {r} </InputItem>
```

Reply:
Then in the tests, you could just loop over the same structure instead of hard-coding in the same tags

Reply:
added in this commit f4a2af5c04f38bfcd41a45343c6fea5ff5fb9872